fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/JsonContainsPath.java

```diff
@@ -0,0 +1,84 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.trees.expressions.functions.scalar;
+
+import org.apache.doris.catalog.FunctionSignature;
+import org.apache.doris.nereids.exceptions.AnalysisException;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.functions.AlwaysNullable;
+import org.apache.doris.nereids.trees.expressions.functions.ExplicitlyCastableSignature;
+import org.apache.doris.nereids.trees.expressions.literal.StringLikeLiteral;
+import org.apache.doris.nereids.trees.expressions.shape.BinaryExpression;
+import org.apache.doris.nereids.trees.expressions.visitor.ExpressionVisitor;
+import org.apache.doris.nereids.types.BooleanType;
+import org.apache.doris.nereids.types.JsonType;
+import org.apache.doris.nereids.types.StringType;
+import org.apache.doris.nereids.types.VarcharType;
+import org.apache.doris.nereids.util.ExpressionUtils;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * ScalarFunction 'json_contains_path'. This class is generated by GenerateFunction .
+ */
+public class JsonContainsPath extends ScalarFunction
+        implements BinaryExpression, ExplicitlyCastableSignature, AlwaysNullable {
+
+    public static final List<FunctionSignature> SIGNATURES = ImmutableList.of(
+            FunctionSignature.ret(BooleanType.INSTANCE)
+                    .varArgs(JsonType.INSTANCE, VarcharType.SYSTEM_DEFAULT, VarcharType.SYSTEM_DEFAULT),
+            FunctionSignature.ret(BooleanType.INSTANCE)
+                    .varArgs(JsonType.INSTANCE, StringType.INSTANCE, StringType.INSTANCE)
+    );
+
+    /**
+     * constructor with 3 arguments or more arguments.
+     */
+    public JsonContainsPath(Expression arg0, Expression arg1, Expression arg2, Expression... varArgs) {
+        super("json_contains_path", ExpressionUtils.mergeArguments(arg0, arg1, arg2, varArgs));
+    }
+
+    /**
+     * withChildren.
+     */
+    @Override
+    public JsonContainsPath withChildren(List<Expression> children) {
+        Preconditions.checkArgument(children.size() >= 3, "json_contains_path should have at least three arguments");
+        return new JsonContainsPath(children.get(0), children.get(1), children.get(2),
+                children.subList(3, children.size()).toArray(new Expression[0]));
+    }
+
+    @Override
+    public List<FunctionSignature> getSignatures() {
+        String oneOrAll = ((StringLikeLiteral) child(1)).getStringValue().toLowerCase();
+        if (!oneOrAll.equals("one") && !oneOrAll.equals("all")) {
+            throw new AnalysisException(
+                    "The oneOrAll argument to json_contains_path may take these values: 'one' or 'all'.");
+        } else {
+            return SIGNATURES;
```

Comment:
if u assume the second arg is `StringLikeLiteral`. u should impl `checkLegalityBeforeTypeCoercion` and check the arg in it